quinn-proto/src/shared.rs

```diff
@@ -257,33 +270,83 @@ pub struct ServerConfig<S>
 where
     S: crypto::Session,
 {
-    /// Transport configuration to use for incoming connections
-    pub transport: Arc<TransportConfig>,
-
-    /// TLS configuration used for incoming connections.
-    ///
-    /// Must be set to use TLS 1.3 only.
+    /// Transport configuration that is used for incoming connections.
+    transport_config: Arc<TransportConfig>,
+    /// Mutable TLS configuration used for incoming connections.
```

Comment:
Why did this lose the sentence about being restricted to TLS 1.3?

Reply:
This is moved to the setter because we should denote restrictions when setting a value. When copying it at two places we will get duplicates.